### PR TITLE
chore: Migrated to android gradle plugin

### PR DESCRIPTION
**Pull request by utpal-barman**
### General

Currently while running on the Android app, it's showing a message on the console to migrate to the Android Gradle Plugin block instead of the deprecated apply script method

```sh
You are applying Flutter's app_plugin_loader Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply

You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply
```

- [Officlal Migration Guideline](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply)

Additionally The latest Flutter 3.19.5 • channel stable — suggests using the newer version of the Gradle version.

To migrate into the Gradle plugins with the declarative plugins block, android platform codes were regenerated by running the `flutter create . platforms=android`, so gradle versions were replaced by the latest flutter version.

#### ⚠️  Blocker and Solution
Currently, there is a blocker to migrate this task, nstack plugin is using an old gradle version without a namespace field in android block, though it was overridden to overcome this issue

### Checklists

- [x] Platform specific changes (input, image picking etc) are tested on both of the platforms (Android and iOS)
- [x] It is tested that the UI changes are rendered correctly on different device sizes (such as long lists or expanded rows) with safe area conditions.
- [x] Edge cases, such as responses being empty or invalid, missing data, no internet connection etc, are tested and the app works as expected.

### Showcase

- [x] Either showcase screenshots / videos are attached, or this PR does not require such any showcase.

### CODE REVIEW

**Comment by hassan-saleh-ml** [comment]
I faced the same blocker with appdynamics on FAB, I tried to override it but failed, I don't remember why it didn't work but maybe I can try it again later today

**Comment by hassan-saleh-ml** [comment]
@utpal-barman did you try to build an apk after these changes?

**Comment by utpal-barman** [comment]
@hassan-saleh-ml 
Yes I tried to build

To make sure again just tried now —

Screenshot 2024-04-23 at 4 05 42 PM


@nivisi Yes, updating NStack is a must now, I saw Hassan already opened a draft PR there https://github.com/nstack-io/flutter-sdk/pull/57

**Review by hassan-saleh-ml — APPROVED** [review]

**Review by nivisi — APPROVED** [review]

File: android/build.gradle [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+                        namespace project.group
+                    }
+                }
+                project.buildscript {
+                    dependencies {
+                        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.10"
+                    }
+                }
+            }
+        }
```
Comment (hassan-saleh-ml) [inline android/build.gradle:29]: The [Nstack PR](https://github.com/nstack-io/flutter-sdk/pull/57) is ready for review, maybe we can wait for it to be merged and remove this workaround

Comment (utpal-barman) [reply android/build.gradle:29]: Ok, i will keep track of this, thanks @hassan-saleh-ml

**Pull request merged**